src/index.js

```diff
@@ -70,7 +78,39 @@ function createReactContext<T>(defaultValue: T): Context<T> {
 
     componentWillReceiveProps(nextProps) {
       if (this.props.value !== nextProps.value) {
-        this.emitter.set(nextProps.value);
+        const oldProps = this.props;
+        const { value: newValue } = nextProps;
+        let changedBits: number;
+        const oldValue = oldProps.value;
+        // Use Object.is to compare the new context value to the old value.
+        // Inlined Object.is polyfill.
+        // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is
+        if (
+          (oldValue === newValue &&
+            (oldValue !== 0 || 1 / oldValue === 1 / newValue)) ||
+          (oldValue !== oldValue && newValue !== newValue) // eslint-disable-line no-self-compare
+        ) {
+          // No change.
+          changedBits = 0;
+        } else {
+          changedBits =
+            typeof calculateChangedBits === 'function'
+              ? calculateChangedBits(oldValue, newValue)
+              : MAX_SIGNED_31_BIT_INT;
+          if (__DEV__) {
```

Comment:
Can you just use `process.env.NODE_ENV !== 'production'` and get rid of all the other stuff to work around `__DEV__`